src/TwigComponent/doc/index.rst

```diff
@@ -337,6 +337,33 @@ You can even give the block default content. See
 :ref:`Passing HTML to Components via Block <embedded-components>`
 for more info.
 
+Composition vs Inheritance
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony tend to follow the best pratice of the component architecture, so it's highly recommended to use Composition instead of Inheritance to reuse code betwen component (see: https://legacy.reactjs.org/docs/composition-vs-inheritance.html).
+
+Let's take an exemple. Some component don't know their children ahead of time. This is common for components like Alert or Dialog that represent generic “boxes”. We recommend you
+to use the block content to pass children directly into their output :
+
+.. code-block:: html+twig
+
+    <div class="alert">
+       {% block content %}
+       {% endblock %}
+    </div>
+
+This lets other components pass arbitrary children to them by nesting the elements :
+
+.. code-block:: html+twig
+
+    <twig:Alert>
+        <twig:Icon name="success"/> 
+        <p>Nice work!</p>
+    </twig:Alert>
+
+Anything inside the <twig:Alert>  tag gets passed into the Alert component as a content block. Since Alert renders {% block content %} inside a <div>, the passed elements appear in the final output.
+You can go deeper on the usage of block, on this part of the :ref:`documentation <embedded-components>`
+
```

Comment:
Does this require a new section ? I feel this is more or less the same as what's already explained in the [Passing HTML to Components Via Blocks](https://symfony.com/bundles/ux-twig-component/current/index.html#passing-html-to-components-via-blocks) section, or did i missread something ?

Reply:
I think so. I feel like this is one of the biggest traps people are falling into. This part is more about what to choose, and the block part is more about how to do it. 
I tried to add more guidelines to the doc